starters/basic-starter/components/PreviewAlert.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+
+export function PreviewAlert() {
+  const router = useRouter()
+  const isPreview = router.isPreview
+  const [showPreviewAlert, setShowPreviewAlert] = useState<boolean>(false)
+
+  useEffect(() => {
+    setShowPreviewAlert(isPreview && window.top === window.self)
+  }, [isPreview])
+
+  if (!showPreviewAlert) {
+    return null
+  }
+
+  return (
+    <div className="sticky top-0 left-0 z-50 w-full px-2 py-1 text-center text-white bg-black">
+      <p className="mb-0">
+        This page is a preview.{" "}
+        <button
+          className="inline-block ml-3 rounded border px-1.5 hover:bg-white hover:text-black active:bg-gray-200 active:text-gray-500"
+          onClick={() => router.push("/api/exit-preview")}
+        >
+          Exit preview mode
+        </button>
```

Comment:
In `preview-alert.tsx` this was:

```
{/* eslint-disable @next/next/no-html-link-for-pages */}
<a href="/api/exit-preview" className="text-white underline">
  Click here
</a>{" "}
to exit preview mode.
```

But the semantics of a link are "take me to a new webpage" and that's not what that does. It's a button because clicking on it performs an action and not a location change. And don't get me started on "click here"!

This is a common accessibility issue, so I've changed element to a button.